File: php/TimestampDataStore.php
<?php
require_once ('MapDataStore.php');

/**
 * Description of TimestampDataStore
 *
 * @author paul
 */
class TimestampDataStore extends MapDataStore {
  public function __construct($type) {
    parent::__construct(strtolower($type) . '-time-to-id-map');
  }
}

?>

File: java/src/main/java/com/caffeinatedbliss/kayveep/TimestampDataStore.java
package com.caffeinatedbliss.kayveep;

import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Map;

/**
 * @author Paul S. Hawke (paul.hawke@gmail.com)
 *         On: 2/26/12 at 12:21 PM
 */
public class TimestampDataStore extends MapDataStore<Long> {
	public TimestampDataStore(String type) {
		super(type.toLowerCase() + "-time-to-id-map");
	}

	@Override
	protected Type getType() {
		return new TypeToken<Map<String, Long>>() {
		}.getType();
	}
}
